File: online-meeting-app-backend/debug_agent_log.js
const fs = require("fs");
const path = require("path");

const LOG_PATH = path.join(__dirname, "..", "debug-fc1f41.log");
const ENDPOINT =
  "http://127.0.0.1:7450/ingest/aaacd22f-394e-4b0b-a706-96fb8d7236b7";

function agentLog(location, message, data = {}, hypothesisId = "") {
  // #region agent log
  const entry = {
    sessionId: "fc1f41",
    location,
    message,
    data,
    hypothesisId,
    timestamp: Date.now(),
  };
  try {
    fs.appendFileSync(LOG_PATH, JSON.stringify(entry) + "\n");
  } catch (_) {}
  fetch(ENDPOINT, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-Debug-Session-Id": "fc1f41",
    },
    body: JSON.stringify(entry),
  }).catch(() => {});
  // #endregion
}

module.exports = { agentLog };

File: AI/debug_agent_log.py
import json
import os
import time

_LOG_PATH = os.path.normpath(
    os.path.join(os.path.dirname(__file__), "..", "debug-fc1f41.log")
)


def agent_log(location, message, data=None, hypothesis_id=""):
    # #region agent log
    try:
        entry = {
            "sessionId": "fc1f41",
            "location": location,
            "message": message,
            "data": data or {},
            "hypothesisId": hypothesis_id,
            "timestamp": int(time.time() * 1000),
        }
        with open(_LOG_PATH, "a", encoding="utf-8") as f:
            f.write(json.dumps(entry) + "\n")
    except Exception:
        pass
    # #endregion
